src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/EnvelopeController.java

```diff
@@ -41,4 +54,26 @@ public EnvelopeMetadataResponse getProcessedEnvelopesByJurisdiction(
             envelopeRetrieverService.getProcessedEnvelopesByJurisdiction(serviceName)
         );
     }
+
+    @PutMapping(path = "/{id}/status")
+    @ApiOperation("Updates the status of given envelope")
+    @ApiResponse(code = 204, message = "Envelope updated")
+    public ResponseEntity<Void> updateStatus(
+        @PathVariable UUID id,
+        @RequestBody StatusUpdate statusUpdate,
+        @RequestHeader(name = "ServiceAuthorization", required = false) String serviceAuthHeader
+    ) {
+        String serviceName = authService.authenticate(serviceAuthHeader);
+
+        // TODO: move to service
+        switch (statusUpdate.status) {
+            case CONSUMED:
+                envelopeUpdateService.markAsConsumed(id, serviceName);
+                break;
+            default:
+                throw new IllegalStateException("No code for handling status: " + statusUpdate.status);
```

Comment:
Due to _new_ enum this branch is now not covered

Reply:
checkstyle complain if `default` is missing

Reply:
`switch` statement has to have `default` - no checkstyle report needed 😝 . for simple usage as it is here, block can be replaced with `if` statement.
since there is only one new state present, there can be zero branches in general and just `markAsConsumed` instantly. but that will introduce risk for future.

so for simple condition test and full branch coverage - just use `if` 👍 . should be all good

Reply:
with `if-elseif-...-else` the question still remains what should go in the last `else`

Reply:
There is no else atm.

In case you want this pattern - re-use existing status and cover this extra branch with unit test.

Reply:
there is no `if` atm. let's talk.